feat: Add Kubewarden resources to the basic and default resourceset

viccuad opened the pull request:
Part of https://github.com/rancher/kubewarden/issues/7.

This PR adds Kubewarden resources to the basic resourceset (without Secrets) and the default resourceset (with Secrets).

This comprises:

- The default Rancher Namespace `catttle-kubewarden-system` (or
  `cattle-kubewarden-*`), and the default Kubewarden Namespace `kubewarden`.
- Kubewarden needed resources installed via the Helm charts.
- Kubewarden CRDs, which get reconciled after restore by the Kubewarden controller.
- The `policy-reporter` subchart of the `kubewarden-controller` chart, for their
  default values. This doesn't include the Grafana integration nor other plugins.

The backup process doesn't include Secrets created to configure PolicyServers
for private registries unless those are correctly labeled by users.

Tested manually by,
applying also https://github.com/rancher/backup-restore-operator/pull/787
and following the steps under https://github.com/kubewarden/docs/pull/632

Documentation in Kubewarden side under:
https://github.com/kubewarden/docs/pull/632

viccuad commented:
Rebased, added missing Namespace backups.

viccuad commented:
Rebased on top of main. Now it adds only to the `basic-resourceset-contents` and `sensitive-resourceset-contents`, given that the `default-resourceset-contents` has been recently dropped.

juadk commented:
Tested and automated by Kubewarden QA here => https://github.com/juadk/helm-charts/actions/runs/16470799097 ✅

viccuad commented:
Fixed the templating problem introduced with adding the optional Value. The resourcesets are under `files/`, which doesn't allow to use go templating. Instead of refactoring and moving all the resource sets under `files/` into `templates/`, I have added a conditional in the templates for `rancher-resourceset-{basic, full}.yaml ` . Please tell me if that's a fitting change.

Renamed the new value to `optionalResources.kubewardenUserCRs.enabled=true` instead of `optionalResources.kubewarden-user-crs.enabled`.

Added 2 new Hull tests for the change.
